.github/workflows/tests.yaml

```diff
@@ -0,0 +1,26 @@
+name: build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    name: Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16.4'
+      - name: make install
+        run: make install
```

Comment:
I think instead of `make install`, we should start with `make vet`. It would also be nice to work in a `make lint` if possible (although I'm not sure this pipeline has golint currently, nor do I know how easy it would be to only lint PR changes)

Reply:
So, I tried to set up linting using `golint` but couldn't figure out how to get it installed using the new actions thing. But, what I _did_ find was `golangcli-lint` which seems to have a lot more attention/support surrounding it https://golangci-lint.run/

With that said, it looks like there are some linting errors in master with this linter that'll prevent this PR from getting in right now. So I figured I'd just check, how do you feel about going with that linter instead? This is the repo for the action I used: https://github.com/golangci/golangci-lint-action

Reply:
Is there any way to make it only check new changes? If not I'm wondering if maybe we shouldn't add a linter to the pipeline until we get our feet under us.  It's a little messy currently :)

Reply:
It does look like there's an `only-new-issues` option here: https://github.com/golangci/golangci-lint-action#how-to-use

But, honestly the linting issues are so minimal (looks like only 6 of them), it's probably worth merging and creating a ticket to fix them IMO. No reason to have potentially unstable code in master, y'know? I could go either way, just my 2 cents

Reply:
So, I went ahead and updated it to where the pipeline will only lint new issues, BUT for now I kept it so that we can see all issues with `make lint`. If you want, I can add in the `--new` flag to the Makefile, but I think it could potentially be misleading/frustrating, since from my understanding, it only understands "new" changes to be uncommitted/unstaged ones. I _believe_ the pipeline linter is smart enough to compare using the `--new-from-rev <REV>` option to compare the PR to master, so we don't have to worry about it there

Reply:
Works for me! We'll see if the linter is indeed smart enough soon... enough

Reply:
Hmmm, I noticed master is failing _but_ if you take a look at the PR build on my branch from these changes it definitely worked. It may only apply to pull requests, so we'll have a constantly failing master until the linting issues get fixed. At least that's my working theory, may be worth putting up a new PR to test

Reply:
When I saw you switched to a different linter, I went and grabbed it myself and fixed the issues locally on another branch I'm working on.  I can probably get it worked out, assuming I ever figure out this dynamic config stuff :(